### ISSUE TITLE
About ViT Backbone

### ISSUE THREAD

**Issue by yunusstalha** [Skyy93/Sample4Geo#36]
In your code you added ViT support like this
`        if "vit" in model_name:
            # automatically change interpolate pos-encoding to img_size
            self.model = timm.create_model(model_name, pretrained=pretrained, num_classes=0, img_size=img_size)`
which I dont understand since  timm.create_model does not have an argument called 'img_size'.  Can you share details on what resolutions you used with ViT ablation test for vigor, and how you implemented these ViTs? Did you used pretrained ViTs ?

**Comment by Skyy93** [comment]
Thanks for reaching out, actually it has this parameter, see:
https://github.com/huggingface/pytorch-image-models/discussions/565

For our ablation in the Sample4Geo we used the pre-trained Vision Transformer from Google: 
https://huggingface.co/timm/vit_base_patch16_224.orig_in21k

and we used the same image sizes as in the ConvNext so for VIGOR it was 384x384 and 384x768. Of course we had to use some pretty ugly solution for the varying image size. We simply cloned the timm repo, and copied the pos_embed and interpolated it. Then in the forward step we had a simple "if length of tensor is XX then use pos_embed1 otherwise pos_embed2". 

I know it is not a nice solution but two individual pos embeddings did the job. 

When I would do it currently I would recommend to use the ViT DINOv2 because of the results we achieved on some similar task:
https://dl.acm.org/doi/abs/10.1145/3689095.3689099
https://dl.acm.org/doi/abs/10.1145/3607834.3616567

If you have further questions just reach out to me :)

EDIT: The upper solution is only necessary if you have weight-shared transformers. Otherwise of course you would simply have two transformers with different pos embeddings.
